keanu-project/src/main/java/io/improbable/keanu/vertices/generic/probabilistic/discrete/CategoricalVertex.java

```diff
@@ -23,6 +25,22 @@
         return new CategoricalVertex<>(defensiveCopy(selectableValues));
     }
 
+    public static <T> CategoricalVertex<T> of(DirichletVertex vertex, List<T> categories) {
+        final int length = ArrayUtil.prod(vertex.getShape());
+        if (length != categories.size()) {
+            throw new IllegalArgumentException("Categories must have length of vertex's size");
+        }
+
+        final int categoriesCount = categories.size();
+
+        return new CategoricalVertex<>(IntStream.range(0, categoriesCount).boxed().collect(toMap(categories::get, i -> new TakeVertex(vertex, 0, i))));
+    }
+
+    public static CategoricalVertex<Integer> of(DirichletVertex vertex) {
+        final int categoriseCount = ArrayUtil.prod(vertex.getShape());
+        return CategoricalVertex.of(vertex, IntStream.range(0, categoriseCount).boxed().collect(Collectors.toList()));
```

Comment:
for clarity could you separate this into more than one line? e.g. 

    categoryNames = IntStream.range(0, categoriseCount)